clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1179,9 +1179,13 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     Builder.defineMacro("__SYCL_DEVICE_ONLY__", "1");
     Builder.defineMacro("SYCL_EXTERNAL", "__attribute__((sycl_device))");
 
-    const llvm::Triple &DeviceTriple = TI.getTriple();
-    if (DeviceTriple.isNVPTX())
-      Builder.defineMacro("__SYCL_NVPTX__", "1");
+    if (TI.getTriple().isNVPTX()) {
+        Builder.defineMacro("__SYCL_NVPTX__", "1");
+    }
+
+    if (TI.getTriple().isAMDGCN()) {
+      Builder.defineMacro("__SYCL_AMDGCN__", "1");
+    }
```

Comment:
A more correct conflict resolution:
```suggestion
    const llvm::Triple &DeviceTriple = TI.getTriple();
    if (DeviceTriple.isNVPTX())
      Builder.defineMacro("__SYCL_NVPTX__", "1");

    if (DeviceTriple.isAMDGCN())
      Builder.defineMacro("__SYCL_AMDGCN__", "1");
```

Reply:
https://github.com/intel/llvm/pull/3977 is removing this part of the code. Can we avoid adding `__SYCL_AMDGCN__`?